Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -847,41 +966,35 @@ protected class ClassWriter : IClassWriter {
 
     protected override void EmitDestructor(string source, Formal dtor, int formalNonGhostIndex, DatatypeCtor ctor,
         List<Type> typeArgs, Type bvType, ConcreteSyntaxTree wr) {
-      string dtorName;
-      if (ctor.EnclosingDatatype is TupleTypeDecl) {
-        throw new NotImplementedException();
-      } else if (int.TryParse(dtor.Name, out _)) {
-        throw new NotImplementedException();
-      } else {
-        dtorName = FormalName(dtor, formalNonGhostIndex);
-      }
-
-      if (ctor.EnclosingDatatype is CoDatatypeDecl) {
-        throw new NotImplementedException();
-      }
-      wr.Write($"(({DtCtorDeclarationName(ctor)}){source}).{dtorName}");
+      wr.Write($"{source}.{IdProtect(dtor.CompileName)}");
```

Comment:
It seems like `EmitDestructor` needs to handle `TupleTypeDecl` too, based on the method comment and the other compilers, EXCEPT for C# for some reason. I can't tell if that's a latent bug for C# or the other compilers have dead code to handle tuples when the method never needs it.

Reply:
Yeah, sorry about that. Fixed it.